src/waveform/waveformwidgetfactory.cpp

```diff
@@ -432,6 +441,22 @@ void WaveformWidgetFactory::setZoomSync(bool sync) {
     }
 }
 
+void WaveformWidgetFactory::setDisplayBeatGrid(bool sync) {
+    m_beatGridEnabled = sync;
+    if (m_config) {
+        m_config->set(ConfigKey("[Waveform]", "beatGridLinesCheckBox"), ConfigValue(m_beatGridEnabled));
```

Comment:
The dlgprefwaveform calls this in order to update the Config
`void DlgPrefWaveform::slotSetGridLines(bool displayGrid) {
    WaveformWidgetFactory::instance()->setDisplayBeatGrid(displayGrid);
}`

Would there be a better way to do this?